yt/frontends/swift/data_structures.py

```diff
@@ -0,0 +1,209 @@
+"""
+Data structures for Swift frontend
+
+
+
+
+"""
+from __future__ import print_function
+
+#-----------------------------------------------------------------------------
+# Copyright (c) 2018, yt Development Team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+
+import numpy as np
+
+from yt.utilities.on_demand_imports import _h5py as h5py
+
+from uuid import uuid4
+
+from yt.utilities.logger import ytLogger as mylog
+
+from yt.frontends.sph.data_structures import \
+    SPHDataset, \
+    SPHParticleIndex
+
+from .fields import SwiftFieldInfo
+
+from yt.data_objects.static_output import \
+    ParticleFile
+
+from yt.funcs import only_on_root
+
+class SwiftDataset(SPHDataset):
+    _index_class = SPHParticleIndex
+    _field_info_class = SwiftFieldInfo
+    _file_class = ParticleFile
+
+    _particle_mass_name = "Masses"
+    _particle_coordinates_name = "Coordinates"
+    _particle_velocity_name = "Velocities"
+    _sph_ptype = "PartType0"
+    _suffix = ".hdf5"
+
+    def __init__(self, filename, dataset_type='swift',
+                 storage_filename=None,
+                 units_override=None):
+
+        self.filename = filename
+
+        super().__init__(filename, dataset_type, units_override=units_override)
+        self.storage_filename = storage_filename
+
+    def _set_code_unit_attributes(self):
+        """
+        Sets the units from the SWIFT internal unit system.
+
+        Currently sets length, mass, time, and temperature.
+
+        Swift outputs in comoving coordinates and no h factors required.
+        """
+        units = self._get_info_attributes("Units")
+
+        if self.cosmological_simulation == 1:
+            only_on_root(
+                    mylog.info, "Assuming length units are in cm (comoving)")
```

Comment:
I'd say `"assuming length units are in comoving centimeters"` if only because `cm` can be understood to mean either comoving or centimeter depending on the context (especially in yt where we overload that in the unit system).